src/prefect/cli/deployment.py

```diff
@@ -874,6 +875,17 @@ async def run(
                 f"\n{available_parameters}"
             )
 
+        if flow_run_name:
+            try:
+                flow_run_name = re.sub(r"{{\s*(\w+)\s*}}", r"{\1}", flow_run_name)
+                flow_run_name = flow_run_name.format(**parameters)
+            except KeyError as e:
+                exit_with_error(
+                    f"Missing parameter for flow run name: '{e.args[0]}' is undefined"
+                )
```

Comment:
It's possible that a deployment has default values for a parameter, which would be a valid reason that a parameter value is missing in this context.

I think it'd be a good idea to read the default parameter values for the associated deployment so that they can be used in flow run name templating.